src/linalg/impl_linalg.rs

```diff
@@ -815,3 +815,132 @@ mod blas_tests {
         assert!(blas_column_major_2d::<f32, _>(&m));
     }
 }
+
+#[allow(dead_code)]
+fn general_outer_to_dyn<Sa, Sb, F, T>(
+    a: &ArrayBase<Sa, IxDyn>,
+    b: &ArrayBase<Sb, IxDyn>,
+    f: F,
+) -> ArrayD<T>
+where
+    T: Copy,
+    Sa: Data<Elem = T>,
+    Sb: Data<Elem = T>,
+    F: Fn(T, T) -> T,
+{
+    //Iterators on the shapes, compelted by 1s
+    let a_shape_iter = a.shape().iter().chain([1].iter().cycle());
+    let b_shape_iter = b.shape().iter().chain([1].iter().cycle());
+
+    let res_ndim = std::cmp::max(a.ndim(), b.ndim());
+    let res_dim: Vec<Ix> = a_shape_iter
+        .zip(b_shape_iter)
+        .take(res_ndim)
+        .map(|(x, y)| x * y)
+        .collect();
+
+    unsafe {
+        let mut res: ArrayD<T> = ArrayBase::uninitialized(res_dim);
+        let res_chunks = res.exact_chunks_mut(b.shape());
+        Zip::from(res_chunks).and(a).apply(|res_chunk, &a_elem| {
+            Zip::from(res_chunk)
+                .and(b)
+                .apply(|res_elem, &b_elem| *res_elem = f(a_elem, b_elem))
+        });
+        res
+    }
+}
+
+#[allow(dead_code, clippy::type_repetition_in_bounds)]
+fn kron_to_dyn<Sa, Sb, T>(a: &ArrayBase<Sa, IxDyn>, b: &ArrayBase<Sb, IxDyn>) -> Array<T, IxDyn>
+where
+    T: Copy,
+    Sa: Data<Elem = T>,
+    Sb: Data<Elem = T>,
+    T: crate::ScalarOperand + std::ops::Mul<Output = T>,
+{
+    general_outer_to_dyn(a, b, std::ops::Mul::mul)
+}
+
+#[allow(dead_code)]
+fn general_outer_same_size<Sa, I, Sb, F, T>(
+    a: &ArrayBase<Sa, I>,
+    b: &ArrayBase<Sb, I>,
+    f: F,
+) -> Array<T, I>
+where
+    T: Copy,
+    Sa: Data<Elem = T>,
+    Sb: Data<Elem = T>,
+    I: Dimension,
+    F: Fn(T, T) -> T,
```

Comment:
Using FnMut is strictly more powerful, and we should do that, unless we have some other goal in mind. Parallelization? Then it needs to be Send/Sync and so on, so maybe it's not that.